scripts/schema/subset_filter.py

```diff
@@ -18,6 +20,45 @@ def filter(fields, subset_file_globs, out_dir):
     if merged_subset:
         fields = extract_matching_fields(fields, merged_subset)
 
+    if excludes:
+        fields = exclude_fields(fields, excludes)
+
+    return fields
+
+
+def pop_field(fields, path):
+    '''pops a field from yaml derived dict using path derived from ordered list of nodes'''
+    node_path = path.copy()
+    if node_path[0] in fields:
+        if len(node_path) == 1:
+            b4 = fields.copy()
+            fields.pop(node_path[0])
+            print("removed field:", (set(b4.keys() ^ set(fields.keys()))).pop())
```

Comment:
I'm finding the set operation here a bit confusing to follow. Is the intent to show the field that was removed?

`.pop()` will remove the value but also returns the element if you want to store it in a variable.

Reply:
Yeah, I didn't use the return value of pop because the field itself can be huge with subfields and comments and all and I didn't want to swamp the output. The set operation was useful to me while debugging but I agree it is OTT now so I'll find a more succinct way, maybe parse the pop return value after all :+1:

Reply:
I've simplified this to take advantage of `flat_name`, test output now looks like:

```
djptek@Elastic-MacBook-Pro ecs % python scripts/generator.py --exclude="./rfcs/text/0017/exclude-set.yml"                (exclude_set)ecs
Loading schemas from local files
Running generator. ECS version 2.0.0-dev
Removed field log.original
djptek@Elastic-MacBook-Pro ecs % python scripts/generator.py --exclude="./rfcs/text/0017/exclude-set-pass2.yml"          (exclude_set)ecs
Loading schemas from local files
Running generator. ECS version 2.0.0-dev
Removed field log.original
Removed field log.file.path
djptek@Elastic-MacBook-Pro ecs % python scripts/generator.py --exclude="./rfcs/text/0017/exclude-set-fail*.yml"          (exclude_set)ecs
Loading schemas from local files
Running generator. ECS version 2.0.0-dev
No match for exclusion: test_fail1_nonexistent_root.file.path
No match for exclusion: test_fail2_nonexistent_node.path
No match for exclusion: path.test_fail3_nonexistent_leaf
```

Reply:
All print statements now removed in favour of unit tests, see: https://github.com/elastic/ecs/pull/1411#issuecomment-841362269